tensorflow/core/kernels/data/shard_dataset_op_test.cc

```diff
@@ -0,0 +1,821 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/kernels/data/dataset_test_base.h"
+
+namespace tensorflow {
+namespace data {
+namespace {
+
+constexpr char kNodeName[] = "shard_dataset";
+constexpr char kOpName[] = "ShardDataset";
```

Comment:
FYI, it would be nice if you could create a separate PR that moves all the dataset op names into a utility namespace `tensorflor::data::name_utils` and replace all uses of string literals for dataset names (e.g. in `DatasetBase::DebugString()` and `IteratorBase::prefix()`) with named constants. If you would like to do that, please send me an initial PR that only does it for a couple of dataset ops so that we can agree on the general approach -- that way you will not have to make O(n) changes based on my feedback.

Reply:
Thanks for your suggestion! Yeah, will work on it.